jaq-native/src/lib.rs

```diff
@@ -260,6 +269,163 @@ fn now() -> Result<f64, Error> {
+    math::math_0_ary!("asinh", libm::asinh, math::as_float, Val::Float),
+    math::math_0_ary!("atan", libm::atan, math::as_float, Val::Float),
+    math::math_0_ary!("atanh", libm::atanh, math::as_float, Val::Float),
+    math::math_0_ary!("cbrt", libm::cbrt, math::as_float, Val::Float),
+    math::math_0_ary!("cos", libm::cos, math::as_float, Val::Float),
+    math::math_0_ary!("cosh", libm::cosh, math::as_float, Val::Float),
+    math::math_0_ary!("erf", libm::erf, math::as_float, Val::Float),
+    math::math_0_ary!("erfc", libm::erfc, math::as_float, Val::Float),
+    math::math_0_ary!("exp", libm::exp, math::as_float, Val::Float),
+    math::math_0_ary!("exp10", libm::exp10, math::as_float, Val::Float),
+    math::math_0_ary!("exp2", libm::exp2, math::as_float, Val::Float),
+    math::math_0_ary!("expm1", libm::expm1, math::as_float, Val::Float),
+    math::math_0_ary!("fabs", libm::fabs, math::as_float, Val::Float),
+    math::math_0_ary!("frexp", libm::frexp, math::as_float, |(m, e)| Val::Arr(
+        Rc::new(vec![Val::Float(m), Val::Int(e as isize)])
+    )),
+    math::math_0_ary!("gamma", libm::tgamma, math::as_float, Val::Float),
+    math::math_0_ary!("j0", libm::j0, math::as_float, Val::Float),
+    math::math_0_ary!("j1", libm::j1, math::as_float, Val::Float),
+    math::math_0_ary!("lgamma", libm::lgamma, math::as_float, Val::Float),
+    math::math_0_ary!("log", libm::log, math::as_float, Val::Float),
+    math::math_0_ary!("log10", libm::log10, math::as_float, Val::Float),
+    math::math_0_ary!("log1p", libm::log1p, math::as_float, Val::Float),
+    math::math_0_ary!("log2", libm::log2, math::as_float, Val::Float),
+    math::math_0_ary!("logb", math::logb, math::as_float, Val::Float),
+    math::math_0_ary!("modf", libm::modf, math::as_float, |(f, r)| Val::Arr(
+        Rc::new(vec![Val::Float(f), Val::Float(r)])
+    )),
+    math::math_0_ary!("nearbyint", libm::round, math::as_float, Val::Float),
+    math::math_0_ary!("pow10", math::pow10, math::as_float, Val::Float),
+    math::math_0_ary!("rint", libm::rint, math::as_float, Val::Float),
+    math::math_0_ary!("significand", math::significand, math::as_float, Val::Float),
+    math::math_0_ary!("sin", libm::sin, math::as_float, Val::Float),
+    math::math_0_ary!("sinh", libm::sinh, math::as_float, Val::Float),
+    math::math_0_ary!("sqrt", libm::sqrt, math::as_float, Val::Float),
+    math::math_0_ary!("tan", libm::tan, math::as_float, Val::Float),
+    math::math_0_ary!("tanh", libm::tanh, math::as_float, Val::Float),
+    math::math_0_ary!("tgamma", libm::tgamma, math::as_float, Val::Float),
+    math::math_0_ary!("trunc", libm::trunc, math::as_float, Val::Float),
+    math::math_0_ary!("y0", libm::y0, math::as_float, Val::Float),
+    math::math_0_ary!("y1", libm::y1, math::as_float, Val::Float),
+    math::math_2_ary!(
+        "atan2",
+        libm::atan2,
+        math::as_float,
+        math::as_float,
+        Val::Float
+    ),
```

Comment:
To make this and following things more compact, I would make some helper macros for the common use case float -> float -> float, int -> float -> float, and float-> int -> float.
Something like `math::fff`, `math::iff`, and `math::fif`.

Reply:
That saves space and makes it much easier to spot differences/typos.